tests/lms/views/decorators/test_h_api.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import json
+import mock
+import pytest
+
+from pyramid.httpexceptions import HTTPBadGateway
+from pyramid.httpexceptions import HTTPBadRequest
+from pyramid.httpexceptions import HTTPGatewayTimeout
+import requests.exceptions
+
+from lms.views.decorators.h_api import create_h_user
+from lms.util import MissingToolConsumerIntanceGUIDError
+from lms.util import MissingUserIDError
+
+
+@pytest.mark.usefixtures("post", "util")
+class TestCreateHUser:
+    def test_it_400s_if_no_oauth_consumer_key_param(self, create_h_user, pyramid_request):
+        del pyramid_request.params["oauth_consumer_key"]
+
+        with pytest.raises(HTTPBadRequest, match="oauth_consumer_key"):
+            create_h_user(pyramid_request, mock.sentinel.jwt)
+
+    def test_it_continues_to_the_wrapped_func_if_feature_not_enabled(self, create_h_user, pyramid_request, wrapped):
+        pyramid_request.params = {"oauth_consumer_key": "foo"}
+
+        returned = create_h_user(pyramid_request, mock.sentinel.jwt)
+
+        wrapped.assert_called_once_with(pyramid_request, mock.sentinel.jwt)
+        assert returned == wrapped.return_value
+
+    def test_it_doesnt_use_the_h_api_if_feature_not_enabled(self, create_h_user, post, pyramid_request):
+        pyramid_request.params = {"oauth_consumer_key": "foo"}
+
+        create_h_user(pyramid_request, mock.sentinel.jwt)
+
+        assert not post.called
```

Comment:
Huh. I've not seen this assertion syntax before. Makes sense.

Reply:
This is actually a difference between the [third-party `mock` library that we're using](http://www.voidspace.org.uk/python/mock/) (`mock`) and [Python 3's standard `unittest.mock` library](https://docs.python.org/3/library/unittest.mock.html) (`unittest.mock`). Back in Python 2 the third-party mock lib was the only mock. For Python 3 that third-party mock lib was moved into the standard lib as `unittest.mock` and extended, but you can still `pip install mock` and use the third-party version in Python 3. The [third-party lib has no `assert_not_called()`](http://www.voidspace.org.uk/python/mock/search.html?q=assert_not_called&check_keywords=yes&area=default) but the [standard lib does have `assert_not_called()`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called).

Since it's Python 3 the lms app should use the new stdlib mock but instead it uses the old third-party one! We should fix that